sources/packages/web/src/services/http/SupportingUserApi.ts

```diff
@@ -13,21 +13,16 @@ export class SupportingUserApi extends HttpBaseClient {
     supportingUserType: SupportingUserType,
     payload: ApplicationIdentifierApiInDTO,
   ): Promise<ApplicationApiOutDTO> {
+    let response;
```

Comment:
The external variable to the `try` does not seems to be needed.
The code can be simplified as below:
```ts
try {
    return await this.postCall(
        this.addClientRoot(`supporting-user/${supportingUserType}/application`),
        payload,
     );
 } catch (error: unknown){
    this.handleAPICustomError(error);
 }
```

Reply:
The issue here is that if the return is inside the try-catch, TS complains because the return can be `undefined`:
![image](https://user-images.githubusercontent.com/78114138/198162481-7de9c413-2c4f-47d1-83b0-c8a21c0a04c0.png)

Reply:
I can see that. All other methods where we use the `handleAPICustomError` are not returning anything, this is the only one that does. The `handleAPICustomError` will throw an exception in one way or another but the compiler is not able to see it.
There is a plan to move the `handleAPICustomError` into the base HTTP class, so, IMO, it is ok to keep it as it is.
Thanks for the explanation.

Reply:
> The issue here is that if the return is inside the try-catch, TS complains because the return can be `undefined`: ![image](https://user-images.githubusercontent.com/78114138/198162481-7de9c413-2c4f-47d1-83b0-c8a21c0a04c0.png)

I think, that because you need to throw the error as in line line 117 of the screenshot
![image](https://user-images.githubusercontent.com/77353155/198345024-7d56e299-e212-4ec3-889e-cb1b436f9cf1.png)

Reply:
Yes @ann-aot the code would be like below. We do not need to bu the compiler is not able to follow.
![image](https://user-images.githubusercontent.com/61259237/198346010-563e47b7-82d3-4e14-8c7a-753b86c3ba48.png)

Reply:
I realize that `hanleAPICustomError`, throws an error, maybe we can add `: never` and it will fix it. devs thoughts? 
![image](https://user-images.githubusercontent.com/77353155/198350018-5930801a-52f4-4110-9b1b-8ecbc600f848.png)